### PR TITLE
Deprecate parameter names in scala.concurrent

### PR DESCRIPTION
**Pull request by viktorklang**
for the purpose of being consistent.
Also switches to Future.successful iso
Promise.successful(..).future for brevity
in implementation code.

### CODE REVIEW

**Comment by adriaanm** [comment]
reviewer?

**Comment by viktorklang** [comment]
Who does reviews for deprecatedNames in the stdlib?

**Comment by soc** [comment]
If this is in line with the decisions on the mailing list (which it seems to be, at least I didn't find a contradiction), it LGTM.

**Comment by viktorklang** [comment]
Yep, I am pretty sure I adhered to the consensus there.

**Comment by soc** [comment]
Ok, great. Yes, LGTM! (Now could you please do this for the rest of the standard library? :-D)

**Comment by adriaanm** [comment]
LGTM echo

Thanks, @soc. 
@viktorklang, Very often, any one of the usual suspects should be able to review. We unfortunately don't have anyone dedicated to the stdlib (we should, but that's another story).

**Comment by heathermiller** [comment]
I thought @axel22 & @phaller were assigned to stdlib, as per the PR policy? Doth quoth the PR policy:

> library: @phaller (Philipp Haller), @axel22 (Aleksander Prokopec -- concurrent & collection)

**Comment by axel22** [comment]
LGTM

File: src/library/scala/concurrent/Future.scala [hunk 1]
```diff
@@ -638,7 +638,7 @@ object Future {
    *  }}}
    */
   def reduce[T, R >: T](futures: TraversableOnce[Future[T]])(op: (R, T) => R)(implicit executor: ExecutionContext): Future[R] = {
-    if (futures.isEmpty) Promise[R]().failure(new NoSuchElementException("reduce attempted on empty collection")).future
+    if (futures.isEmpty) failed(new NoSuchElementException("reduce attempted on empty collection"))
```
Comment (phaller) [inline src/library/scala/concurrent/Future.scala:641]: Please incorporate feedback from previous PR https://github.com/scala/scala/pull/2509

File: src/library/scala/concurrent/Future.scala [hunk 2]
```diff
@@ -592,15 +592,15 @@ object Future {
...
+    val futuresBuffer = futures.toBuffer
+    if (futuresBuffer.isEmpty) successful[Option[T]](None)
     else {
       val result = Promise[Option[T]]()
-      val ref = new AtomicInteger(futures.size)
+      val ref = new AtomicInteger(futuresBuffer.size)
       val search: Try[T] => Unit = v => try {
         v match {
-          case Success(r) => if (predicate(r)) result tryComplete Success(Some(r))
+          case Success(r) if p(r) => result tryComplete Success(Some(r))
```
Comment (phaller) [inline src/library/scala/concurrent/Future.scala:603]: Like in previous PR https://github.com/scala/scala/pull/2509

``` scala
    result trySuccess Some(r)
```

**Comment by viktorklang** [comment]
Feedback incorporated!

**Pull request merged**